packages/core-data/src/actions.js

```diff
@@ -166,35 +166,37 @@ export function receiveEmbedPreview( url, preview ) {
  * @param {Object}   [options]                 Delete options.
  * @param {Function} [options.__unstableFetch] Internal use only. Function to
  *                                             call instead of `apiFetch()`.
- *                                             Must return a control descriptor.
+ *                                             Must return a promise.
  */
-export function* deleteEntityRecord(
+export const deleteEntityRecord = (
 	kind,
 	name,
 	recordId,
 	query,
-	{ __unstableFetch = null } = {}
-) {
-	const entities = yield getKindEntities( kind );
+	{ __unstableFetch = triggerFetch } = {}
+) => async ( { dispatch } ) => {
+	const entities = await dispatch( getKindEntities( kind ) );
 	const entity = find( entities, { kind, name } );
 	let error;
 	let deletedRecord = false;
 	if ( ! entity ) {
 		return;
 	}
 
-	const lock = yield* __unstableAcquireStoreLock(
-		STORE_NAME,
-		[ 'entities', 'data', kind, name, recordId ],
-		{ exclusive: true }
+	const lock = await dispatch(
+		__unstableAcquireStoreLock(
+			STORE_NAME,
+			[ 'entities', 'data', kind, name, recordId ],
+			{ exclusive: true }
+		)
 	);
 	try {
-		yield {
+		await dispatch( {
```

Comment:
This is a sync action that merely updates a `deleting[ recordId ].pending` flag in the store.

Reply:
Hm the more I think about it, the more I am on board with not `await`-ing `dispatch` calls – only reducer is at play here and so nothing should surprise us. I'll remove this await and the other, similar one 👍